rust/datafusion/tests/sql.rs

```diff
@@ -94,6 +94,31 @@ fn parquet_query() {
     assert_eq!(expected, actual);
 }
 
+#[test]
+fn parquet_partition_query() {
+    let mut ctx = ExecutionContext::new();
+    let testdata = env::var("PARQUET_TEST_DATA").expect("PARQUET_TEST_DATA not defined");
+    ctx.register_parquet("alltypes_plain", &format!("{}/partition/", testdata))
+        .unwrap();
+    let sql = "SELECT id, string_col FROM alltypes_plain";
+    execute(&mut ctx, sql);
+}
```

Comment:
Partition support is only available with the new physical query plan and not with the direct execution from the logical plan (which will be deprecated once the physical query execution is a little farther along - see https://issues.apache.org/jira/browse/ARROW-5227 for more info).

Try this instead though:

```
    let plan = ctx.create_logical_plan(&sql).unwrap();
    let plan = ctx.optimize(&plan).unwrap();
    let plan = ctx.create_physical_plan(&plan, DEFAULT_BATCH_SIZE).unwrap();
    let results = ctx.collect(plan.as_ref()).unwrap();
    for batch in results {
        println!("{} x {}", batch.num_rows(), batch.num_columns());
    }
```

Reply:
@andygrove I've created a new ticket for a bug using the snippet above: https://issues.apache.org/jira/browse/ARROW-6702